doc/ADR/canonical_goto.md

```diff
@@ -0,0 +1,22 @@
+\page canonical-goto Canonical goto definition
+
+During the analysis of a program CBMC transforms the program in an intermediate language called
+extended goto. Then the tool performs a series of simplifications on the obtained goto program until
+the program is given to the solver.
+
+Here we define the concept of **canonical goto program** as a program that has been simplified to
+remove all extended-goto specific features.
+
+More specifically, a program that is in **canonical goto program form** is one that can be analyzed
+by any solver deriving from `goto_verifiert` by using the `goto_verifiert::operator()` function
+without requiring any lowering step.
+
+At the time of writing verifiers extending `goto_verifiert`, so defining what a **canonical goto
```

Comment:
I can't parse this sentence. Also, this definition of canonical form doesn't really help me much. I'd rather have
a description than defining it implicitly as whatever these verifiers can process. If I wanted to write something of my own that took the canonical form this wouldn't help me much.

Reply:
Having a high-level, declarative specification of what GOTO is and of the rules checked by the verifiers would solve the underlying problem of helping people build a mental model of canonical GOTO independently of the C++ that implements the concept.